### PR TITLE
DATA-190 CI Master workflow

### PR DESCRIPTION
**Pull request by adrien-mg**
* Added CI master workflow,
* Dusting Makefile.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,72 +8,62 @@ MAKEFLAGS += --no-builtin-rules
...
 	rm -rf dist
 	rm -rf build
-	rm -rf tmp
 	rm -rf $(PKG).egg-info
-.PHONY: clean
 
-## tmp/.sentinel.install-venv                        : installs virtual env
-tmp/.sentinel.install-venv: requirements.txt setup.py
+## check     : runs checks
+check: .sentinel.unit-tests .sentinel.integration-tests
```
Comment (graham-jackson) [inline Makefile:32]: I'm not sure what integration tests the tap will end up having, but I don't mind keeping a placeholder for them 👍

**Review by graham-jackson — APPROVED** [review]
Minor comments and questions.

Pre-approving.

File: .env.sample [hunk 2]
```diff
@@ -0,0 +1 @@
+PYPISERVER_URL=
```
[the remark this answers is not in the document]
Comment (adrien-mg) [reply .env.sample:1]: Yeah good point, it's not too sensitive I guess :shrug:

File: .github/workflows/master.yaml [hunk 3]
```diff
@@ -1,30 +1,23 @@
...
     env:
       PYPISERVER_URL: ${{ secrets.PYPISERVER_URL }}
-
     steps:
-      - name: Install python3.7
-        id: install-python
-        run: |
-          sudo add-apt-repository ppa:deadsnakes/ppa
-          sudo apt-get update
-          sudo apt-get install python3.7 python3.7-dev python3.7-venv
```
[the remark this answers is not in the document]
Comment (adrien-mg) [reply .github/workflows/master.yaml:22]: I don't really know how it got there in the first place :thinking: We don't need that as it's already installed on the worker instance, I think this is a remnant from using Github's own Action workers, before we set up our own Action worker.

File: Makefile [hunk 4]
```diff
@@ -8,72 +8,62 @@ MAKEFLAGS += --no-builtin-rules
...
+## clean     : removes sentinel files
 clean:
+	rm -f .sentinel*
+
+## clean-hard: removes sentinel files, venv, dist, build and egg dirs
+clean-hard: clean
 	rm -rf $(VENV)
 	rm -rf dist
 	rm -rf build
-	rm -rf tmp
```
[the remark this answers is not in the document]
Comment (adrien-mg) [reply Makefile:33]: Absolutely.

File: Makefile [hunk 5]
```diff
@@ -8,72 +8,62 @@ MAKEFLAGS += --no-builtin-rules
...
+## check     : runs checks
+check: .sentinel.unit-tests .sentinel.integration-tests
+
+## build     : builds the package
+build: .sentinel.build
+
+## upload    : uploads the package to a remote pypi server
+upload: .sentinel.upload
+
+# private
```
[the remark this answers is not in the document]
Comment (adrien-mg) [reply Makefile:40]: Ouh, sorry it's an abstract thing I like, I see the documented targets as mostly phony and the public part of the Makefile. Anything that uses sentinel files, or that are intermediate steps, I kind of consider them private, not sure how sound that is :shrug:

**Pull request merged**